packages/ref-metrics-pmp/src/cmip_ref_metrics_pmp/annual_cycle.py

```diff
@@ -0,0 +1,194 @@
+from collections.abc import Iterable
+
+from cmip_ref_core.datasets import FacetFilter, SourceDatasetType
+from cmip_ref_core.metrics import CommandLineMetric, DataRequirement, MetricExecutionDefinition, MetricResult
+from cmip_ref_metrics_pmp.pmp_driver import build_pmp_command, process_json_result
+
+
+class AnnualCycle(CommandLineMetric):
+    """
+    Calculate the annual cycle for a dataset
+    """
+
+    def __init__(self) -> None:
+        self.name = "PMP Annual Cycle"
+        self.slug = "pmp-annual-cycle"
+        self.data_requirements = (
+            DataRequirement(
+                source_type=SourceDatasetType.obs4MIPs,
+                filters=(FacetFilter(facets={"source_id": ("GPCP-2-3"), "variable_id": ("pr")}),),
```

Comment:
Is this on Obs4MIPs or in your registry?

Reply:
I see GPCP is marked as "published on ESGF" on https://github.com/orgs/Climate-REF/projects/3, but I also have some PMP ref data that can be fetched for annual cycle. What would be the best approach?

Reply:
Let's use your annual cycle data for now until we have a better solution

Reply:
Here is the list of PMP ref datasets for annual cycle. Could you please help implementing fetching mechanism? 

https://pcmdi.llnl.gov/pss/pmpdata/PMP_obs4MIPsClims_v20250228.txt

Reply:
Sure. This requires #219

Reply:
The working branch of this PR is rebased to the latest main which includes changes from #219.